src/orthonormal_bases.jl

```diff
@@ -343,3 +343,39 @@ function get_basis(M::Manifold, x, B::ProjectedOrthonormalBasis{:gram_schmidt,
     @warn "get_basis with bases $(typeof(B)) only found $(K) orthonormal basis vectors, but manifold dimension is $(dim)."
     return PrecomputedOrthonormalBasis(Ξ)
 end
+
+function show(io::IO, ::ArbitraryOrthonormalBasis{F}) where {F}
+    print(io, "ArbitraryOrthonormalBasis($(F))")
+end
+function show(io::IO, ::ProjectedOrthonormalBasis{method,F}) where {method,F}
+    print(io, "ProjectedOrthonormalBasis($(method), $(F))")
+end
+function show(io::IO, mime::MIME"text/plain", onb::DiagonalizingOrthonormalBasis)
+    println(
+        io,
+        "DiagonalizingOrthonormalBasis with scalars in $(number_system(onb)) and 0 curvature in direction:",
+    )
+    show(io, mime, onb.v)
+end
+function show(io::IO, mime::MIME"text/plain", onb::PrecomputedOrthonormalBasis)
+    println(io, "PrecomputedOrthonormalBasis with scalars in $(number_system(onb))")
+    print(io, "Basis vectors:")
+    for v in onb.vectors
```

Comment:
I think we may want to limit the number of vectors that are shown here. Maybe something like 5?

Reply:
Yes, I think for `ProductManifold`, `ProductGroup`, the product representations, and the bases, it would probably be best to truncate at some point. I'm thinking showing the first 2 and last 2 with dots in the middle if the number is beyond something like 5.

Reply:
Yes, that's a good approach.